api/image_pipeline/pipeline_image_util.py

```diff
@@ -0,0 +1,660 @@
+        bundle_component = get_bundle_override(distgit_repo_name, version)
+        if not bundle_component:
+            bundle_component = f"{'-'.join(brew_name.split('-')[:-1])}-metadata-component"
+        bundle_distgit = f"{distgit_repo_name}-bundle"
+
+        brew_object.bundle_component = bundle_component
+        brew_object.bundle_distgit = bundle_distgit
+
+    # Tag
+    tag = get_image_stream_tag(distgit_repo_name, version)
+    if tag:
+        brew_object.payload_tag = tag
+
+    return github_object, distgit_object, brew_object
+
+
+def get_brew_id(brew_name: str) -> int:
+    """
+    Get the brew id for the given brew name.
+
+    :so: SlackOutput object for reporting results.
+    :brew_name: The name of the brew package
+    """
+    try:
+        koji_api = util.koji_client_session()
+    except Exception:
+        raise exceptions.KojiClientError("Failed to connect to Brew.")
+
+    try:
+        brew_id = koji_api.getPackageID(brew_name, strict=True)
+    except Exception:
+        raise exceptions.BrewIdNotFound(f"Brew ID not found for brew package `{brew_name}`. Check API call.")
+
+    return brew_id
+
+
+def brew_to_cdn(brew_name: str, variant_name: str) -> list:
+    """
+    Function to return all the Brew to CDN mappings (since more than one could be present)
+
+    :brew_name: Brew package name
+    :variant_name: The name of the product variant eg: 8Base-RHOSE-4.10
+    """
+    url = f"https://errata.devel.redhat.com/api/v1/cdn_repo_package_tags?filter[package_name]={brew_name}"
+    response = request_with_kerberos(url)
+
+    repos = []
+    for item in response.json()['data']:
+        repos.append(item['relationships']['cdn_repo']['name'])
+
+    repos = list(set(repos))  # Getting only the unique repo names
+
+    # Cross-check to see if the repo is mapped to the given variant
+    results = []
+    for repo in repos:
+        response = get_cdn_repo_details(repo)
+        for variant in response['data']['relationships']['variants']:
+            if variant['name'] == variant_name:
+                results.append(repo)
+                break
+
+    if not results:
+        raise exceptions.CdnFromBrewNotFound(f"CDN was not found for brew `{brew_name}` and variant `{variant_name}`")
+    return results
+
+
+def brew_to_delivery(brew_package_name: str, variant: str, brew_object) -> None:
+    """
+    Driver function for Brew -> Delivery pipeline
+
+    :brew_package_name: Brew package name
+    :variant: The 8Base-RHOSE variant
+    """
+    cdn_repo_names = brew_to_cdn(brew_package_name, variant)
+
+    for cdn_repo_name in cdn_repo_names:
+        # CDN
+        cdn_object = get_cdn_payload(cdn_repo_name, variant)
+
+        # CDN -> Delivery
+        cdn_object.delivery = cdn_to_delivery_payload(cdn_repo_name)
+
+        brew_object.cdn += [cdn_object]
+
+
+# @util.cached
+def doozer_brew_distgit(version: str) -> list:
+    output = util.cmd_gather(
```

Comment:
I like to see named tuples here, e.g.
```suggestion
    _, output, stderr = util.cmd_gather(
```
which leaves no doubt about which you're looking at below

Reply:
Fixed!